posthog/api/feature_flag.py

```diff
@@ -121,9 +129,112 @@ def get_queryset(self) -> QuerySet:
+        else:
+            posthoganalytics.capture(
+                request.user.distinct_id, self._analytics_updated_event_name,
+            )
+        return feature_flag_override
+
+    def update(self, instance: FeatureFlagOverride, validated_data: Dict) -> FeatureFlagOverride:
+        request = self.context["request"]
+        posthoganalytics.capture(
+            request.user.distinct_id, self._analytics_updated_event_name,
+        )
+        return super().update(instance, validated_data)
+
+
+class FeatureFlagOverrideViewset(
+    AnalyticsDestroyModelMixin, viewsets.ModelViewSet,
+):
+    queryset = FeatureFlagOverride.objects.all()
+    serializer_class = FeatureFlagOverrideSerializer
+    permission_classes = [IsAuthenticated, ProjectMembershipNecessaryPermissions]
+    authentication_classes = [
+        PersonalAPIKeyAuthentication,
+        TemporaryTokenAuthentication,
+        authentication.SessionAuthentication,
+        authentication.BasicAuthentication,
+    ]
+
+    def get_queryset(self) -> QuerySet:
+        if not self.request.user.is_authenticated:  # for mypy, since 'AnonymousUser' has no 'team'
+            raise exceptions.NotAuthenticated
+        queryset = (
+            super()
+            .get_queryset()
+            .filter(feature_flag__team=self.request.user.team, user__current_team=self.request.user.team)
+        )
+        if self.action == "list":
+            feature_flag_id = self.request.GET.get("feature_flag_id")
+            if feature_flag_id:
+                queryset = queryset.filter(feature_flag_id=feature_flag_id)
+        return queryset
+
+    def perform_create(self, serializer):
+        self._check_team_permissions(serializer)
+        return super().perform_create(serializer)
+
+    def perform_update(self, serializer):
+        self._check_team_permissions(serializer)
+        return super().perform_update(serializer)
+
+    def _check_team_permissions(self, serializer: FeatureFlagOverrideSerializer):
+        if not self.request.user.is_authenticated:  # for mypy, since 'AnonymousUser' has no 'team'
+            raise exceptions.NotAuthenticated
+
+        feature_flag = serializer.validated_data.get("feature_flag")
+        user = serializer.validated_data.get("user")
+
+        if serializer.instance:
+            exisiting_feature_flag_override = cast(FeatureFlagOverride, serializer.instance)
+            if not feature_flag and exisiting_feature_flag_override:
+                feature_flag = exisiting_feature_flag_override.feature_flag
+            if not user and exisiting_feature_flag_override:
+                user = exisiting_feature_flag_override.user
+
+        if feature_flag.team != self.request.user.team or user.team != self.request.user.team:
+            raise PermissionDenied
+
+    @action(methods=["GET"], detail=False)
+    def my_overrides(self, request: request.Request, **kwargs):
+        queryset = super().get_queryset().filter(user=request.user)
```

Comment:
Not sure if there's a potential data leak here. I could have a feature flag override but then be removed from the team/org, but I would still see the FF.